code/game/objects/items/miningradio.dm

```diff
@@ -0,0 +1,174 @@
+#define WEATHER_ALERT_CLEAR 0
+#define WEATHER_ALERT_INCOMING 1
+#define WEATHER_ALERT_IMMINENT_OR_ACTIVE 2
+
+/// Portable mining radio purchasable by miners
+/obj/item/radio/weather_monitor
+	icon = 'icons/obj/miningradio.dmi'
+	name = "mining weather radio"
+	icon_state = "miningradio"
+	desc = "A weather radio designed for use in inhospitable environments. Gives audible warnings when storms approach. Has access to cargo channel."
+	freqlock = RADIO_FREQENCY_LOCKED
+
+/obj/item/radio/weather_monitor/Initialize(mapload)
+	. = ..()
+	AddComponent( \
+		/datum/component/weather_announcer, \
+		state_normal = "weatherwarning", \
+		state_warning = "urgentwarning", \
+		state_danger = "direwarning", \
+	)
+	set_frequency(FREQ_SUPPLY)
+
+/// Wall mounted mining weather tracker
+/obj/machinery/mining_weather_monitor
+	name = "barometric monitor"
+	desc = "A machine monitoring atmospheric data from mining environments. Provides warnings about incoming weather fronts."
+	icon = 'icons/obj/miningradio.dmi'
+	icon_state = "wallmount"
+
+/obj/machinery/mining_weather_monitor/Initialize(mapload, ndir, nbuild)
+	. = ..()
+	AddComponent( \
+		/datum/component/weather_announcer, \
+		state_normal = "wallgreen", \
+		state_warning = "wallyellow", \
+		state_danger = "wallred", \
+	)
+
+/obj/machinery/mining_weather_monitor/update_overlays()
+	. = ..()
+	if((machine_stat & BROKEN) || !powered())
+		return
+	. += emissive_appearance(icon, "emissive", src)
+
+/// Component which makes you yell about what the weather is
+/datum/component/weather_announcer
```

Comment:
Put this in its own file, past Jacquerel only didn't do this before downstream because he was lazy.

Reply:
will do!